cli/index.ts

```diff
@@ -0,0 +1,126 @@
+
+export function getLink(
+  space: string,
+  filters: Record<string, string>,
+  columns: string[],
+): string {
+  const kibanaFilters = Object.entries(filters).map(([key, value]) => {
+    return `(query:(match_phrase:(${key}:'${value}')))`;
+  });
+
+  // The `#/` at the end is important for Kibana to correctly parse the query string
+  // The `URL` object moves this to the end of the string, which breaks the link.
+  const base = `https://logs.gutools.co.uk/s/${space}/app/discover#/`;
+
+  const query = {
+    ...(kibanaFilters.length > 0 && {
+      _g: `(filters:!(${kibanaFilters.join(",")}))`,
+    }),
+    ...(columns.length > 0 && {
+      _a: `(columns:!(${columns.join(",")}))`,
+    }),
+  };
+
+  const queryString = Object.entries(query)
+    .map(([key, value]) => `${key}=${value}`)
+    .join("&");
+
+  return `${base}?${queryString}`;
+}
+
+function parseArguments(args: string[]): Args {
+  return parse(args, {
+    boolean: ["follow"],
+    negatable: ["follow"],
+    string: ["space", "stack", "stage", "app"],
+    collect: ["column", "filter"],
+    stopEarly: false,
+    "--": true,
+    default: {
+      follow: true,
+      column: ["message"],
+      space: "default",
+      filter: [],
+    },
+  });
+}
+
+function escapeColon(str: string): string {
+  return str.includes(":") ? `'${str}'` : str;
+}
+
+function parseFilters(filter: string[]): Record<string, string> {
+  return filter.reduce((acc, curr) => {
+    const [key, value] = curr.split("=");
+    return { ...acc, [escapeColon(key)]: value };
+  }, {});
+}
+
+function removeUndefined(
+  obj: Record<string, string | undefined>,
+): Record<string, string> {
+  return Object.entries(obj).filter(([, value]) => !!value).reduce(
+    (acc, [key, value]) => ({
+      ...acc,
+      [key]: value,
+    }),
+    {},
+  );
+}
+
+function printHelp(): void {
+  console.log(`Usage: devx-logs [OPTIONS...]`);
+  console.log("\nOptional flags:");
+  console.log("  --help              Display this help and exit");
+  console.log("  --space             The Kibana space to use");
+  console.log("  --stack             The stack tag to filter by");
+  console.log("  --stage             The stage tag to filter by");
+  console.log("  --app               The app tag to filter by");
+  console.log(
+    "  --column            Which columns to display. Multiple: true. Default: 'message'",
+  );
+  console.log(
+    "  --filter            Additional filters to apply. Multiple: true. Format: key=value",
+  );
+  console.log("  --no-follow         Don't open the link in the browser");
+  console.log("\nExample:");
+  console.log(
+    "  devx-logs --space devx --stack deploy --stage PROD --app riff-raff",
+  );
+  console.log("\nAdvanced example:");
+  console.log(
+    "  devx-logs --space devx --stack deploy --stage PROD --app riff-raff --filter level=INFO --filter region=eu-west-1 --column message --column logger_name",
+  );
+}
```

Comment:
This looks a bit untidy... blame `deno fmt` 😉.

Reply:
I think https://deno.land/x/flags_usage@2.0.0 handles the `--help` command  for you so you don't have to type it manually and keep it up to date if you make changes to the CLI

Reply:
Whilst I agree that manually implementing this help message isn't great, and prone to becoming stale, I don't anticipate it changing much.

The linked `flags_usage` package was last published 2 years ago. We don't have a defined vulnerability scanning story for the Deno ecosystem yet, so tempted to avoid bringing in third-party dependencies until absolutely needed.

I'll look at replacing usage of https://deno.land/x/deno_open@v0.0.6 with the standard library.